### PR TITLE
PhysicsTools fixes for Muon Tag & Probe in 92X

### PR DESCRIPTION
**Pull request by gpetruc**
Small fixes to allow running muon tag & probe in 92X
 * remove references to `AssociationVector` of `bool` from the `ValueMap` converter: it's not needed, and throws in 92X due to missing dictionary.
 * tell `PATTriggerProducer` to not depend on the `TriggerResults` of its own process (fixes errors when running in scheduled mode)

tested with 10024.0, 136.731, 1330.0

@HuguesBrun

### CODE REVIEW

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by perrotta** [comment]
+1
It runs
No visible effect on jenkins outputs

File: PhysicsTools/PatAlgos/plugins/PATTriggerProducer.cc [hunk 1]
```diff
@@ -157,7 +157,7 @@ PATTriggerProducer::PATTriggerProducer( const ParameterSet & iConfig ) :
 
   // HLT configuration parameters
   if ( iConfig.exists( "triggerResults" ) ) tagTriggerResults_ = iConfig.getParameter< InputTag >( "triggerResults" );
-  triggerResultsGetter_ = GetterOfProducts< TriggerResults >( InputTagMatch( InputTag( tagTriggerResults_.label(), tagTriggerResults_.instance() ) ), this);
+  triggerResultsGetter_ = GetterOfProducts< TriggerResults >( InputTagMatch( InputTag( tagTriggerResults_.label(), tagTriggerResults_.instance(), autoProcessName_ ? InputTag::kSkipCurrentProcess : nameProcess_  ) ), this);
```
Comment (Dr15Jones) [inline PhysicsTools/PatAlgos/plugins/PATTriggerProducer.cc]: I do not think this works as intended. InputTagMatch does an exact match on the process name if one is given:
https://github.com/cms-sw/cmssw/blob/master/FWCore/Framework/interface/InputTagMatch.h#L38

I'm afraid no process name will ever match the string associated to `InputTag::kSkipCurrentProcess`.

Comment (Dr15Jones) [inline PhysicsTools/PatAlgos/plugins/PATTriggerProducer.cc]: I believe I have changed the algorithm to do what was intended: #19259

**Pull request merged**